packages/babel-generator/src/node/parentheses.ts

```diff
@@ -301,6 +301,10 @@ export function SequenceExpression(
   if (parentType === "ClassDeclaration" || parentType === "ClassExpression") {
     return parent.superClass === node;
```

Comment:
It seems to me that if a SequenceExpression is an immediate child of Class, it must be the superClass, since the other productions are wrapped within either class element or the Decorator node.

If that's the case, then the `ClassExpression` branch can also be removed as it is already covered by the last return.